### ISSUE TITLE
Squid data?

### ISSUE THREAD

**Issue by iruindegi** [firehol/netdata#283]
Hi,

Really a great job!! but I was thinking... is there any plugin for squid?

**Comment by Ubiquitous** [comment]
It's at the link below:

[/charts.d/squid.chart.sh](https://github.com/firehol/netdata/blob/8bb28f2f6a6526847fc3d2bc8a216e041a0f8677/charts.d/squid.chart.sh)

**Comment by iruindegi** [comment]
great!!
but mmmmhhhh how do I enable it?

**Comment by iruindegi** [comment]
oh! I found this https://github.com/firehol/netdata/tree/8bb28f2f6a6526847fc3d2bc8a216e041a0f8677/charts.d but when I followed the instructions I have this error:

```
tail -f /var/log/netdata/* | grep squid
. Please set squid_url='url' and squid_host='host' and squid_port='port' in /etc/netdata/squid.conf
charts.d: chart 'squid' check() function reports failure.
. Please set squid_url='url' and squid_host='host' and squid_port='port' in /etc/netdata/squid.conf
charts.d: chart 'squid' check() function reports failure.
```

This is my `/etc/netdata/squid.conf` file:
```
squid_host=127.0.0.1
squid_port=3128
squid_url="cache_object://127.0.0.1:3128/counters"
squid_timeout=2
squid_update_every=5
```

any help or clue?

**Comment by Ubiquitous** [comment]
Those are the default settings, so you'll want to change them to reflect your squid settings.

**Comment by ktsaou** [comment]
the plugin executes this:

```sh
squidclient -h $host -p $port $url
```

Try it by hand. If you can do it on the command line, netdata will be able to do it too.

**Comment by iruindegi** [comment]
~# squidclient -h 127.0.0.1 -p 3128 cache_object://127.0.0.1:3128/counters
-bash: squidclient: no se encontró la orden

so I need to install squidclient

**Comment by iruindegi** [comment]
I did some progress

```
root@machine:~# squidclient -h 127.0.0.1 -p 3128 cache_object://127.0.0.1:3128/counters
HTTP/1.0 200 OK
Date: Thu, 21 Apr 2016 06:26:45 GMT
Server: Apache/2.2.16 (Debian)
Last-Modified: Wed, 30 Apr 2014 08:52:52 GMT
ETag: "2d73a-21-4f83eac774100"
Accept-Ranges: bytes
Content-Length: 33
Content-Type: text/html
Age: 4827
X-Cache: HIT from machine
X-Cache-Lookup: HIT from machine:3128
Via: 1.1 machine:3128 (squid/2.7.STABLE9)
Connection: close

<html>
<body>


</body>
</html>

root@machine:~# killall netdata
root@machine:~# /usr/sbin/netdata
root@machine:~# tail -f /var/log/netdata/* | grep squid
```

Now I have this error:


. Please set squid_url='url' and squid_host='host' and squid_port='port' in /etc/netdata/squid.conf
charts.d: chart 'squid' check() function reports failure.

But something is wrong because I´d defined variables...

**Comment by ktsaou** [comment]
The response you get from your squid is not right.
This is mine:

```sh
 $ squidclient -h 127.0.0.1 -p 3128 cache_object://127.0.0.1:3128/cou                                                                nters
HTTP/1.1 200 OK
Server: squid/3.5.15
Mime-Version: 1.0
Date: Thu, 21 Apr 2016 08:27:21 GMT
Content-Type: text/plain;charset=utf-8
Expires: Thu, 21 Apr 2016 08:27:21 GMT
Last-Modified: Thu, 21 Apr 2016 08:27:21 GMT
X-Cache: MISS from box
X-Cache-Lookup: MISS from box:3128
Connection: close

sample_time = 1461227229.272067 (Thu, 21 Apr 2016 08:27:09 GMT)
client_http.requests = 269926
client_http.hits = 11579
client_http.errors = 3706
client_http.kbytes_in = 108336
client_http.kbytes_out = 4144999
client_http.hit_kbytes_out = 272855
server.all.requests = 143654
server.all.errors = 0
server.all.kbytes_in = 3623066
server.all.kbytes_out = 91348
server.http.requests = 143615
server.http.errors = 0
server.http.kbytes_in = 3622790
server.http.kbytes_out = 91313
server.ftp.requests = 0
server.ftp.errors = 0
server.ftp.kbytes_in = 0
server.ftp.kbytes_out = 0
server.other.requests = 39
server.other.errors = 0
server.other.kbytes_in = 275
server.other.kbytes_out = 34
icp.pkts_sent = 0
icp.pkts_recv = 0
icp.queries_sent = 0
icp.replies_sent = 0
icp.queries_recv = 0
icp.replies_recv = 0
icp.query_timeouts = 0
icp.replies_queued = 0
icp.kbytes_sent = 0
icp.kbytes_recv = 0
icp.q_kbytes_sent = 0
icp.r_kbytes_sent = 0
icp.q_kbytes_recv = 0
icp.r_kbytes_recv = 0
icp.times_used = 0
cd.times_used = 0
cd.msgs_sent = 0
cd.msgs_recv = 0
cd.memory = 0
cd.local_memory = 1081
cd.kbytes_sent = 0
cd.kbytes_recv = 0
unlink.requests = 10974
page_faults = 284
select_loops = 11900641
cpu_time = 1017.703000
wall_time = 12.440148
swap.outs = 21593
swap.ins = 10146
swap.files_cleaned = 0
aborted_requests = 3033
```

**Comment by iruindegi** [comment]
maybe is because we use squidguard?

**Comment by ktsaou** [comment]
I forgot to answer this.
I think it has nothing to do with squidguard.

Have you solved it?

**Comment by josealgardataborges** [comment]
@ktsaou any way to prevent squid3 from login the netdata request for cache_object?

**Comment by paulfantom** [comment]
I believe preventing requests is solved in #548.

@iruindegi have you solved it?

**Comment by ktsaou** [comment]
Cleaning up and closing this. If there is something pending please post and I'll re-open it.
